storage/libradb/src/state_store/mod.rs

```diff
@@ -77,24 +77,25 @@ impl StateStore {
         let (new_root_hash_vec, tree_update_batch) =
             JellyfishMerkleTree::new(self).put_blob_sets(blob_sets, first_version)?;
 
-        cs.counter_bumps.bump(
-            LedgerCounter::NewStateNodes,
-            tree_update_batch.node_batch.len(),
-        );
-        cs.counter_bumps.bump(
-            LedgerCounter::NewStateLeaves,
-            tree_update_batch.num_new_leaves,
-        );
+        let num_versions = new_root_hash_vec.len();
+        assert_eq!(num_versions, tree_update_batch.node_stats.len());
+
+        tree_update_batch
+            .node_stats
+            .iter()
+            .enumerate()
+            .for_each(|(i, stats)| {
+                let counter_bumps = cs
+                    .counter_bumps
+                    .entry(first_version + i as u64)
+                    .or_insert_with(LedgerCounterBumps::new);
```

Comment:
This is repeated so much I'm thinking properly we should let LedgerCounterBumps or another type to manage an internal hash map, or at least short cut this call (`let mut bumps = cs.counter_bumps.for_version(v)`)